resources/exec.go

```diff
@@ -485,3 +513,27 @@ func (w *wrapWriter) Write(p []byte) (int, error) {
 func (w *wrapWriter) String() string {
 	return w.Buffer.String()
 }
+
+// cmdFiles returns all the potential files/commands this command might need
+func (obj *ExecRes) cmdFiles() []string {
+	var paths []string
+	if obj.Shell != "" {
+		paths = append(paths, obj.Shell)
+	} else {
+		cmdSplit := strings.Fields(obj.Cmd)
+		paths = append(paths, cmdSplit[0])
```

Comment:
the logic is right in that we won't get an empty "" because obj.Cmd is in Validate() but it could be a " " space, so we still need to check we have at least `len(cmdSplit) > 0` to avoid a panic. Similarly, logic is good below but needs these checks I think.

Reply:
fixed